lucene/core/src/java/org/apache/lucene/codecs/VectorWriter.java

```diff
@@ -194,6 +197,7 @@ public int nextDoc() throws IOException {
       current = docIdMerger.next();
       if (current == null) {
         docId = NO_MORE_DOCS;
+        size = ord;
```

Comment:
Maybe also add a comment here so we understand the sneakiness in context?

Reply:
Thanks, yes I did.